java/core/src/main/java/com/lancedb/lance/ipc/ScanOptions.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.ipc;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Optional;
+import org.apache.arrow.util.Preconditions;
+
+/**
+ * Lance scan options.
+ */
+public class ScanOptions {
+  private final Optional<Integer> fragmentId;
```

Comment:
could we ever support fragmentIds ?

Reply:
Yeah we can support that, i will add that func